.github/workflows/CodeAnalysisAndCI.yml

```diff
@@ -36,7 +36,6 @@ jobs:
           token: ${{ secrets.CODECOV_TOKEN }}
           flags: unittests # optional
           name: jacoco-report-uploader # optional
-          fail_ci_if_error: true # optional (default = false)
```

Comment:
Should it not fail on error?

Reply:
I thought it best to leave codecov as an informational part of the pipeline as it has some quirks which can be annoying. For example, if I change the name of a variable in a test it would fail the pipeline as code coverage would not have increased.
Let me know your thoughts please @t92549

Reply:
Yeah that is probably fair it does get annoying